cmake/external/xnnpack.cmake

```diff
@@ -24,43 +24,67 @@ set(onnxruntime_EXTERNAL_LIBRARIES_XNNPACK XNNPACK pthreadpool)
+  endfunction()
+
+  if(onnxruntime_ENABLE_WEBASSEMBLY_THREADS)
     target_compile_options(XNNPACK PRIVATE "-pthread")
   endif()
 
+  GetSrcListFromBazel("PROD_SCALAR_WASM_MICROKERNEL_SRCS" prod_scalar_wasm_srcs)
+  GetSrcListFromBazel("ALL_WASM_MICROKERNEL_SRCS" all_wasm_srcs)
+  GetSrcListFromBazel("WASM32_ASM_MICROKERNEL_SRCS" wasm32_asm_srcs)
+
+  message(DEBUG "prod_scalar_wasm_srcs: ${prod_scalar_wasm_srcs}\n")
+  message(DEBUG "all_wasm_srcs: ${all_wasm_srcs}\n")
+  message(DEBUG "wasm32_asm_srcs: ${wasm32_asm_srcs}\n")
+
   message("Adding WebAssembly Source Files to XNNPACK")
-  set(wasm_src_patterns "${XNNPACK_DIR}/src/init.c"
-                        "${XNNPACK_DIR}/src/params-init.c"
-                        "${XNNPACK_DIR}/src/qu8-avgpool/9*-minmax-scalar-c1.c"
-                        "${XNNPACK_DIR}/src/u8-rmax/scalar.c"
-                        "${XNNPACK_DIR}/src/x8-zip/x*-scalar.c"
-                        "${XNNPACK_DIR}/src/x*-transposec/gen/2x4-scalar-int.c"
-                        "${XNNPACK_DIR}/src/wasm-*.c"
-                        "${XNNPACK_DIR}/src/*-wasm-*.c"
-                        "${XNNPACK_DIR}/src/*-wasm.c")
-  set(wasm32_asm_src_patterns "${XNNPACK_DIR}/src/wasm_shr_*.S")
-
-  file(GLOB_RECURSE XNNPACK_WASM_MICROKERNEL_SRCS CONFIGURE_DEPENDS ${wasm_src_patterns})
-  file(GLOB_RECURSE XNNPACK_WASM32_ASM_MICROKERNEL_SRCS CONFIGURE_DEPENDS ${wasm32_asm_src_patterns})
-
-  message(DEBUG "XNNPACK_WASM_MICROKERNEL_SRCS:${XNNPACK_WASM_MICROKERNEL_SRCS}")
-  message(DEBUG "XNNPACK_WASM32_ASM_MICROKERNEL_SRCS:${XNNPACK_WASM32_ASM_MICROKERNEL_SRCS}")
-
-  target_sources(XNNPACK PRIVATE ${XNNPACK_WASM_MICROKERNEL_SRCS}
-                                 ${XNNPACK_WASM32_ASM_MICROKERNEL_SRCS})
-
-  if (onnxruntime_ENABLE_WEBASSEMBLY_SIMD)
-    target_compile_options(XNNPACK PRIVATE "-msimd128")
+  set(wasm_srcs "")
+  list(APPEND wasm_srcs ${prod_scalar_wasm_srcs})
+  list(APPEND wasm_srcs ${all_wasm_srcs})
+  list(APPEND wasm_srcs ${wasm32_asm_srcs})
 
-    set(wasmsimd_src_patterns "${XNNPACK_DIR}/src/wasmsimd-*.c"
-                              "${XNNPACK_DIR}/src/*-wasmsimd-*.c"
-                              "${XNNPACK_DIR}/src/*-wasmsimd.c"
-                              "${XNNPACK_DIR}/src/*/wasmsimd.c")
+  target_sources(XNNPACK PRIVATE ${wasm_srcs})
 
-    file(GLOB_RECURSE XNNPACK_WASMSIMD_MICROKERNEL_SRCS CONFIGURE_DEPENDS ${wasmsimd_src_patterns})
-    message(DEBUG "XNNPACK_WASMSIMD_MICROKERNEL_SRCS:${XNNPACK_WASMSIMD_MICROKERNEL_SRCS}")
+  if(onnxruntime_ENABLE_WEBASSEMBLY_SIMD)
+    GetSrcListFromBazel("ALL_WASMSIMD_MICROKERNEL_SRCS" all_wasmsimd_srcs)
+    message(DEBUG "all_wasmsimd_srcs: ${all_wasmsimd_srcs}")
 
-    target_sources(XNNPACK PRIVATE ${XNNPACK_WASMSIMD_MICROKERNEL_SRCS})
+    target_compile_options(params_init PRIVATE "-msimd128")
+    target_compile_options(XNNPACK PRIVATE "-msimd128")
+    target_sources(XNNPACK PRIVATE ${all_wasmsimd_srcs})
```

Comment:
Need to set this on params_init as well to get __wasm_simd128__ defined so the wasm simd params are included in the build of params-init.c